exporter/fileexporter/file_exporter.go

```diff
@@ -142,109 +39,41 @@ type Exporter struct {
 }
 
 func (e *Exporter) ConsumeTraces(_ context.Context, td pdata.Traces) error {
-	octds := internaldata.TraceDataToOC(td)
-	for _, octd := range octds {
-		// Ensure only one write operation happens at a time.
-		e.mutex.Lock()
-		defer e.mutex.Unlock()
-
-		// Prepare to write JSON object.
-		jw := &jsonWriter{writer: e.file}
-		if err := jw.Begin(); err != nil {
-			return err
-		}
-		defer jw.End()
-
-		if err := exportResourceAndNode(jw, octd.Node, octd.Resource); err != nil {
+	for _, rs := range pdata.TracesToOtlp(td) {
+		if err := exportMessageAsLine(e, rs); err != nil {
 			return err
 		}
-
-		if err := jw.BeginMarshalArray("spans"); err != nil {
-			return err
-		}
-		defer jw.EndMarshalArray()
-
-		for _, span := range octd.Spans {
-			if span != nil {
-				if err := jw.MarshalArrayItem(span); err != nil {
-					return err
-				}
-			}
-		}
 	}
 	return nil
 }
 
 func (e *Exporter) ConsumeMetrics(_ context.Context, md pdata.Metrics) error {
-	ocmds := pdatautil.MetricsToMetricsData(md)
-	for _, ocmd := range ocmds {
-		// Ensure only one write operation happens at a time.
-		e.mutex.Lock()
-		defer e.mutex.Unlock()
-
-		// Prepare to write JSON object.
-		jw := &jsonWriter{writer: e.file}
-		if err := jw.Begin(); err != nil {
-			return err
-		}
-		defer jw.End()
-
-		if err := exportResourceAndNode(jw, ocmd.Node, ocmd.Resource); err != nil {
+	for _, rs := range data.MetricDataToOtlp(pdatautil.MetricsToInternalMetrics(md)) {
+		if err := exportMessageAsLine(e, rs); err != nil {
 			return err
 		}
+	}
+	return nil
+}
 
-		if err := jw.BeginMarshalArray("metrics"); err != nil {
+func (e *Exporter) ConsumeLogs(_ context.Context, ld pdata.Logs) error {
+	for _, rs := range pdata.LogsToOtlp(ld) {
```

Comment:
Instead of printing one item per line convert to the "Request" message and print the entire message:
```golang
request := &otlptrace.ExportTraceServiceRequest{
	ResourceSpans: pdata.TracesToOtlp(td),
}
exportMessageAsLine(e, request)
```

I think it is better this way.

Reply:
Did that for traces and metrics, could not find request for logs.

Reply:
LogsToOtlp?

Reply:
Oops, now I found it